core/src/main/scala/org/http4s/Uri.scala

```diff
@@ -374,6 +377,129 @@ object Uri {
       case n => b.setLength(n)
     }
 
+  /**
+    * Taken from https://github.com/scalatra/rl/blob/v0.4.10/core/src/main/scala/rl/UrlCodingUtils.scala
+    * Copyright (c) 2011 Mojolly Ltd.
+    */
+  private[http4s] val Unreserved =
+    CharPredicate.AlphaNum ++ "-_.~"
+
+  private val toSkip =
+    Unreserved ++ "!$&'()*+,;=:/?@"
+
+  // scalastyle:off magic.number
+  private val HexUpperCaseChars = (0 until 16).map { i =>
+    Character.toUpperCase(Character.forDigit(i, 16))
+  }
+  // scalastyle:on magic.number
+
+  /**
+    * Percent-encodes a string.  Depending on the parameters, this method is
+    * appropriate for URI or URL form encoding.  Any resulting percent-encodings
+    * are normalized to uppercase.
+    *
+    * @param toEncode the string to encode
+    * @param charset the charset to use for characters that are percent encoded
+    * @param spaceIsPlus if space is not skipped, determines whether it will be
+    * rendreed as a `"+"` or a percent-encoding according to `charset`.
+    * @param toSkip a predicate of characters exempt from encoding.  In typical
+    * use, this is composed of all Unreserved URI characters and sometimes a
+    * subset of Reserved URI characters.
+    */
+  def urlEncode(
```

Comment:
Bikeshed: why (other than private historical reasons) `url` in a class called `Uri`?  Maybe just `encode`?  Similar for `decode`.

Reply:
You are right! Done.